fbgemm_gpu/codegen/training/forward/embedding_forward_split_kernel_template.cu

```diff
@@ -115,26 +115,24 @@ using namespace fbgemm_gpu;
     // Iterate over the row in the weights table, in 4-element strides
     #pragma unroll kMaxVecsPerThread
     for (int32_t i = 0;
-        i < kMaxVecsPerThread && (i * kThreadGroupSize + threadIdx.x) * VEC_WIDTH < D;
+        i < kMaxVecsPerThread;// && (i * kThreadGroupSize + threadIdx.x) * VEC_WIDTH < D;
         ++i) {
         // Load the slice of the weights
-        const int32_t d = (i * kThreadGroupSize + threadIdx.x) * VEC_WIDTH;
+        //const int32_t d = (i * kThreadGroupSize + threadIdx.x) * VEC_WIDTH;
+        int32_t d = (i * kThreadGroupSize + threadIdx.x) * VEC_WIDTH;
+        d = (d < D) ? d : 0;
 
-        {%- if is_gwd_kernel %}
-        auto weights_slice = weights_row.load(d, qparams);
-        // Scale weights with global weight decay
-        weights_slice.mul_(global_weight_decay_j);
-        {%- else %}
+        //Vec4T<cache_t> dummyv;
+        //const auto weights_slice = (d<D) ? weights_row.load(d, qparams) : dummyv;
         const auto weights_slice = weights_row.load(d, qparams);
-        {%- endif %}
-
-        {%- if weighted %}
+        vals[_j*kMaxVecsPerThread + i] = weights_slice;
+        //{%- if weighted %}
```

Comment:
Consider removing them if they are not useful anymore.
If it's some upcoming feature, we can add them back later.

Reply:
Done